### ISSUE TITLE
The 3.2.3-1 version failed to compile on Debian 9 (the package installation is also invalid). The debug verification environment is backed up.

### ISSUE THREAD

**Issue by cliusz** [shadowsocks/shadowsocks-libev#2278]
### What version of shadowsocks-libev are you using?
3.2.3-1

### What operating system are you using?
Debian 9 x64
Linux libev 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux

### What did you do?
root@libev:~# history
    0  cd
    1  apt update
    2  apt upgrade
    3  apt install sudo
    4  apt install git

development environment:
    5  sudo apt-get install --no-install-recommends gettext build-essential autoconf libtool libpcre3-dev asciidoc xmlto libev-dev libc-ares-dev automake libmbedtls-dev libsodium-dev

libsodium:
    6  export LIBSODIUM_VER=1.0.17
    7  wget https://download.libsodium.org/libsodium/releases/libsodium-$LIBSODIUM_VER.tar.gz
    8  tar xvf libsodium-$LIBSODIUM_VER.tar.gz
    9  pushd libsodium-$LIBSODIUM_VER
   10  ./configure --prefix=/usr && make
   11  sudo make install
   12  popd
   13  sudo ldconfig

mbedtls:
   14  export MBEDTLS_VER=2.16.0
   15  wget https://tls.mbed.org/download/mbedtls-$MBEDTLS_VER-gpl.tgz
   16  tar xvf mbedtls-$MBEDTLS_VER-gpl.tgz
   17  pushd mbedtls-$MBEDTLS_VER
   18  make SHARED=1 CFLAGS=-fPIC
   19  sudo make DESTDIR=/usr install
   20  popd
   21  sudo ldconfig

download source of libev:
   22  git clone https://github.com/shadowsocks/shadowsocks-libev.git
   23  cd shadowsocks-libev
   24  git submodule update --init --recursive

build own deb packages:
   25  sudo sh -c 'printf "deb http://deb.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/stretch-backports.list'
   26  sudo apt-get install --no-install-recommends devscripts equivs
   27  mk-build-deps --root-cmd sudo --install --tool "apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends -y"
   28  ./autogen.sh && dpkg-buildpackage -b -us -uc

ERROR appears Here!
       dpkg-checkbuilddeps: error: Unmet build dependencies: libsodium-dev (>= 1.0.12) pkg-config

   29  cd ..
   30  sudo dpkg -i shadowsocks-libev*.deb

Configure and start the service  ......

### What did you expect to see?
Successfully compiled, installed, and configured to take effect.
NOTE: The installation of the official various other methods on Debian 9 did not work.

### What did you see instead?
dpkg-buildpackage: info: source package shadowsocks-libev
dpkg-buildpackage: info: source version 3.2.3-1
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Max Lv <max.c.lv@gmail.com>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build shadowsocks-libev
dpkg-checkbuilddeps: error: Unmet build dependencies: libsodium-dev (>= 1.0.12) pkg-config
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)
------ from dpkg-buildpackage -b -us -uc

### What is your config in detail (with all sensitive info masked)?
In summary, I would like to ask about the relevant problems and their solutions.
VPS server debug verification is available if needed. Thank you very much!

**Comment by cliusz** [comment]
Previous "27 mk-build-deps" show result below:
dh_builddeb
dpkg-deb: building package 'shadowsocks-libev-build-deps' in '../shadowsocks-libev-build-deps_3.2.3-1_all.deb'.

The package has been created.
Attention, the package has been created in the current directory,
not in ".." as indicated by the message above!
Selecting previously unselected package shadowsocks-libev-build-deps.
(Reading database ... 34458 files and directories currently installed.)
Preparing to unpack shadowsocks-libev-build-deps_3.2.3-1_all.deb ...
Unpacking shadowsocks-libev-build-deps (3.2.3-1) ...
Reading package lists... Done
Building dependency tree
Reading state information... Done
Correcting dependencies...Starting pkgProblemResolver with broken count: 1
Starting 2 pkgProblemResolver with broken count: 1
Investigating (0) shadowsocks-libev-build-deps:amd64 < 3.2.3-1 @iU mK Nb Ib >
Broken shadowsocks-libev-build-deps:amd64 Depends on libsodium-dev:amd64 < 1.0.11-2 @ii mK > (>= 1.0.12)
  Removing shadowsocks-libev-build-deps:amd64 because I can't find libsodium-dev:amd64
Done
 Done
Starting pkgProblemResolver with broken count: 0
Starting 2 pkgProblemResolver with broken count: 0
Done
The following packages will be REMOVED:
  shadowsocks-libev-build-deps
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 9,216 B disk space will be freed.
(Reading database ... 34462 files and directories currently installed.)
Removing shadowsocks-libev-build-deps (3.2.3-1) ...
mk-build-deps: Unable to install shadowsocks-libev-build-deps at /usr/bin/mk-build-deps line 402.
mk-build-deps: Unable to install all build-dep packages

**Comment by cliusz** [comment]
I have successfully installed libsodium-dev 1.0.17 before command 13, and why the command 27 mk-build-deps can't search it. How should I explicitly specify the library path?

**Comment by cliusz** [comment]
Even if libmbedtls-dev libsodium-dev is not installed in command 5, the error message in command 27 remains the same (prefix only before 1.0.11-2 none | 1.0.11-2):

Correcting dependencies...Starting pkgProblemResolver with broken count: 1
Starting 2 pkgProblemResolver with broken count: 1
Investigating (0) shadowsocks-libev-build-deps:amd64 < 3.2.3-1 @iU mK Nb Ib >
Broken shadowsocks-libev-build-deps:amd64 Depends on libsodium-dev:amd64 < none | 1.0.11-2 @un uH > (>= 1.0.12)
   Removing shadowsocks-libev-build-deps:amd64 because I can't find libsodium-dev:amd64

why?
How can I specify the libsodium installed earlier?

Here, I would like to ask the teachers, how can I compile the package in debian 9 to install shadowsocks-libev 3.2.3?
Looking forward to your guidance! Thank you very much!
